c/src/ml-api-remote-service.c

```diff
@@ -154,6 +155,120 @@ _mlrs_get_service_type (gchar * service_str)
   return service_type;
 }
 
+/**
+ * @brief Get ml remote service activation type.
+ */
+static gboolean
+_mlrs_parse_activate (gchar * activate)
+{
+  gboolean ret = TRUE;
+
+  if (activate && g_ascii_strcasecmp (activate, "false") == 0) {
+    ret = FALSE;
+  }
+
+  return ret;
+}
+
+/**
+ * @brief Callback function for receving data using curl.
+ */
+static size_t
+curl_mem_write_cb (void *data, size_t size, size_t nmemb, void *clientp)
+{
+  size_t recv_size = size * nmemb;
+  GByteArray *array = (GByteArray *) clientp;
+
+  if (!array)
+    return 0;
+
+  g_byte_array_append (array, data, recv_size);
+
+  return recv_size;
+}
+
+/**
+ * @brief Register model file given by the remote sender.
+ */
+static gboolean
+_mlrs_model_register (gchar * service_key, nns_edge_data_h data_h,
+    void *data, nns_size_t data_len)
+{
+  guint version = -1;
+  gchar *description = NULL;
+  gchar *name = NULL;
+  gchar *current_dir = g_get_current_dir ();
+  gchar *dir_path = NULL;
+  gchar *model_path = NULL;
+  gchar *activate = NULL;
+  gboolean active_bool = TRUE;
+  GError *error = NULL;
+  gboolean ret = TRUE;
+
+  nns_edge_data_get_info (data_h, "description", &description);
+  nns_edge_data_get_info (data_h, "name", &name);
+  nns_edge_data_get_info (data_h, "activate", &activate);
+  active_bool = _mlrs_parse_activate (activate);
+
+  dir_path = g_build_path ("/", current_dir, service_key, NULL);
+  g_mkdir_with_parents (dir_path, 0755);
+  model_path = g_build_path ("/", dir_path, name, NULL);
+
+  if (!g_file_set_contents (model_path, (char *) data, data_len, &error)) {
+    _ml_loge ("Failed to write data to file: %s",
+        error ? error->message : "unknown error");
+    g_clear_error (&error);
+    ret = FALSE;
+    goto error;
+  }
+
+  /**
+   * @todo Hashing the path. Where is the default path to save the model file?
+   */
+  ml_service_model_register (service_key, model_path,
+      active_bool, description, &version);
```

Comment:
```  ret = ml_service_model_register (service_key, model_path,  active_bool, description, &version); ``` , right?

Reply:
Changed to check the return value. Thanks.